Test Failing

**Describe the bug**
One test is failing. If I just rerun the test, it successes sometimes:

FAILED tests/test_executor.py::test_run_only_some_steps 

Eg:- https://github.com/stanford-crfm/marin/pull/585

**To Reproduce**

[(Steps to reproduce the behavior:)](https://github.com/stanford-crfm/marin/pull/585)

--- a/tests/test_executor.py
+++ b/tests/test_executor.py
@@ -349,7 +349,7 @@ def test_run_only_some_steps():
         executor = create_executor(temp_dir)
         executor.run(steps=[a, b, c], run_only=["a", "c"])
 
+        # these can execute in any order
         results = read_log(log)
         assert len(results) == 2
-        assert results[0] is None
-        assert results[1]["m"] == 10
+        assert (results[0] is None and results[1]["m"] == 10) or (results[1] is None and results[0]["m"] == 10)

